cdap-docs/build.sh

```diff
@@ -20,283 +20,210 @@
-    run_command docs-github-part ${source_path}
-  elif [ "${doc_type}" == "${WEB}" -o "${doc_type}" == "${WEB_ONLY}" ]; then
-    run_command docs-web-part ${source_path}
-  fi
-  set_display_version
-  display_messages_file
+}
+
+function build_docs_set() {
+  local warnings
+  local title="Building Docs Set"
+  display_start_title ${title}
+  
+  clean_targets
+  clear_messages_file
+  build_docs_first_pass
+  clear_messages_file
+  build_javadocs docs
+  build_docs_cli
+  build_docs_second_pass
+  build_docs_package
+  
+  set_and_display_version
+  display_messages
   warnings=$?
   cleanup_messages_file
-  echo "--------------------------------------------------------"
-  ring_bell "Completed build of \"${doc_type}\""
-  echo "========================================================"
-  echo "========================================================"
-  echo
-  return ${warnings}
+  display_end_title_bell ${title}
+  exit ${warnings}
+}
+
+function build_docs_only() {
+  local warnings
+  local title="Building Docs Only"
+  display_start_title ${title}
+  
+  clear_messages_file
+  build_docs_second_pass
+  
+  set_and_display_version
+  display_messages
+  warnings=$?
+  cleanup_messages_file
+  display_end_title_bell ${title}
+  exit ${warnings}
+}
+
+function build_docs_first_pass() {
+  local title="Building Docs First Pass"
+  display_start_title ${title}
+
+  build_docs_inner_level build-docs
+  
+  display_end_title ${title}
+}
+
+function build_docs_second_pass() {
+  local title="Building Docs Second Pass"
+  display_start_title ${title}
+
+  build_docs_inner_level build-web
+  build_docs_outer_level ${GOOGLE_TAG_MANAGER_CODE}
+  copy_docs_inner_level
+
+  display_end_title ${title}
 }
 
 function build_javadocs() {
   local javadoc_type=${1}
-  echo "========================================================"
-  echo "Building Javadocs: '${javadoc_type}'"
-  echo "--------------------------------------------------------"
-  echo
-  if [ ${NO_JAVADOCS} ]; then
+  local title="Building Javadocs: '${javadoc_type}'"
+  display_start_title ${title}
+
+  local warnings
+  if [[ -n ${NO_JAVADOCS} ]]; then
     echo_red_bold "Javadocs disabled."
   else
-    build_javadocs_api ${javadoc_type}
     USING_JAVADOCS="true"
     export USING_JAVADOCS
+    if [ "${javadoc_type}" == "${ALL}" ]; then
+      local javadoc_run="mvn javadoc:aggregate -P release"
+    else
+      local javadoc_run="mvn clean site -P templates"
+    fi
+    if [ "${DEBUG}" == "${TRUE}" ]; then
+      local debug_flag="-X"
+    else
+      local debug_flag=''
+    fi
+    set_mvn_environment
+    echo "JAVA_HOME: ${JAVA_HOME}"
+    echo "JAVA Version:"
+    java -version
+    echo "Maven Version:"
+    mvn -version
+    local start=`date`
+    echo "Maven clean install"
+    MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean install -P examples,templates,release -DskipTests -Dgpg.skip=true
```

Comment:
I think the two runs are required, and when we attempted to put them together it failed. The thought was this was a consequence of the dependencies in CDAP, that two runs were required; the product had to be built first so that the different modules could link together correctly. Sort of like the docs with a multi-manual project: two runs are required there to enable the index to be correctly linked.